api/provider/aws/service_read.go

```diff
@@ -54,13 +77,72 @@ func (s *ServiceProvider) readService(clusterName, serviceID string) (*ecs.Servi
 	})
 
 	output, err := s.AWS.ECS.DescribeServices(input)
+	if err != nil || len(output.Services) == 0 {
+		return nil, errors.Newf(errors.ServiceDoesNotExist, "Service '%s' does not exist")
+	}
+
+	return output.Services[0], nil
+}
+
+func (s *ServiceProvider) populateModelTags(serviceID string, model *models.Service) error {
+	tags, err := s.TagStore.SelectByTypeAndID("service", serviceID)
 	if err != nil {
-		return nil, err
+		return err
+	}
+
+	if tag, ok := tags.WithKey("environment_id").First(); ok {
+		model.EnvironmentID = tag.Value
+	}
+
+	if tag, ok := tags.WithKey("load_balancer_id").First(); ok {
+		model.LoadBalancerID = tag.Value
 	}
 
-	for _, service := range output.Services {
-		return service, nil
+	if tag, ok := tags.WithKey("name").First(); ok {
+		model.ServiceName = tag.Value
 	}
 
-	return nil, fmt.Errorf("ecs service '%s' in cluster '%s' does not exist", serviceID, clusterName)
+	if model.EnvironmentID != "" {
+		tags, err := s.TagStore.SelectByTypeAndID("environment", model.EnvironmentID)
+		if err != nil {
+			return err
+		}
+
+		if tag, ok := tags.WithKey("name").First(); ok {
+			model.EnvironmentName = tag.Value
+		}
+	}
+
+	if model.LoadBalancerID != "" {
+		tags, err := s.TagStore.SelectByTypeAndID("load_balancer", model.LoadBalancerID)
+		if err != nil {
+			return err
+		}
+
+		if tag, ok := tags.WithKey("name").First(); ok {
+			model.LoadBalancerName = tag.Value
+		}
+	}
+
+	deployments := []models.Deployment{}
+	for _, deploy := range model.Deployments {
+		tags, err := s.TagStore.SelectByTypeAndID("deploy", deploy.DeployID)
+		if err != nil {
+			return err
+		}
+
+		if tag, ok := tags.WithKey("name").First(); ok {
+			deploy.DeployName = tag.Value
+		}
+
+		if tag, ok := tags.WithKey("version").First(); ok {
+			deploy.DeployVersion = tag.Value
+		}
+
+		deployments = append(deployments, deploy)
```

Comment:
This seems odd, I assume it's a carryover from the previous implementation. Can't we just update these in place instead of re-creating all of the deployments?

Reply:
✅